### PR TITLE
Migrate Maven Feed Resource

### PR DESCRIPTION
**Pull request by IsaacCalligeros95**
Note: This change branches off of the testing changes, one that has merged this change is ~10 files, **Not worth reviewing until then.**

I've added a migration test `maven_feed_resource_migration_test`. This test sets the `TF_CLI_CONFIG_FILE` to avoid any terraformrc configurations to ensure we run against an older version of the provider prior to the mux server changes.

These tests run a configuration against the old provider, assert the properties match as expected, then run a plan check. The plan check asserts that the resulting plan is empty, i.e. for the same configuration run against the SDK provider we get the same result on the updated plugin provider. Then we update the resource, assert all properties have updated correctly, delete the resource and assert is has been deleted as expected.

**Maven Feed** 
```
resource "octopusdeploy_maven_feed" "maven_newz" {
    name = "maven feed new tf Z"
    feed_uri = "http://localhosty:8066"
    space_id = "Spaces-1"
}
```

Before migration:
image

After:
image

**Project group schema changes:**
```
resource "octopusdeploy_project_group" "project_group" {
      description = "Financial Applications"
      name        = "Finance z z z"
}
```

Before:
image

After
image

### CODE REVIEW

File: octopusdeploy_framework/schemas/schema_maven_feed.go
Comment (IsaacCalligeros95) [inline octopusdeploy_framework/schemas/schema_maven_feed.go:32]: These were set on the old schema, but are not respected. The API just returns ["Server", "ExecutionTarget"]. This can be dropped but I've kept it here for consistency with the old schema.

**Comment by benPearce1** [comment]
@IsaacCalligeros95 I tested this locally and got some oddness around the username.

```
resource "octopusdeploy_maven_feed" "maven" {
    name = "maven feed"
    feed_uri = "http://localhost"
    space_id = octopusdeploy_space.space.id
}
```

>╷
│ Error: Provider produced inconsistent result after apply
│ 
│ When applying changes to octopusdeploy_maven_feed.maven, provider "provider[\"octopus.com/com/octopusdeploy\"]" produced an unexpected new value: .username: inconsistent values for sensitive attribute.

It also seems to be missing the original implementation validation for username and password being empty:
this is from the 0.21.1


>│ Error: expected "username" to not be an empty string, got 
│ 
│   with octopusdeploy_maven_feed.maven,
│   on main.tf line 29, in resource "octopusdeploy_maven_feed" "maven":
│   29:     username = ""
│ 
╵
╷
│ Error: expected "password" to not be an empty string, got 
│ 
│   with octopusdeploy_maven_feed.maven,
│   on main.tf line 30, in resource "octopusdeploy_maven_feed" "maven":
│   30:     password = ""
│ 
╵

**Comment by IsaacCalligeros95** [comment]
> > │ Error: Provider produced inconsistent result after apply
> > │
> > │ When applying changes to octopusdeploy_maven_feed.maven, provider "provider["octopus.com/com/octopusdeploy"]" produced an unexpected new value: .username: inconsistent values for sensitive attribute.

Sorry, I missed this on this PR when backporting the fix for this. On the upgrade calls I have to check the Username is not empty.
> 	if feed.Username != "" {
		data.Username = types.StringValue(feed.Username)
	}



> 
> It also seems to be missing the original implementation validation for username and password being empty: this is from the 0.21.1
> 
> > │ Error: expected "username" to not be an empty string, got
> > │
> > │   with octopusdeploy_maven_feed.maven,
> > │   on main.tf line 29, in resource "octopusdeploy_maven_feed" "maven":
> > │   29:     username = ""
> > │
> > ╵
> > ╷
> > │ Error: expected "password" to not be an empty string, got
> > │
> > │   with octopusdeploy_maven_feed.maven,
> > │   on main.tf line 30, in resource "octopusdeploy_maven_feed" "maven":
> > │   30:     password = ""
> > │
> > ╵

I've added the validation to this change and will merge that into the other branches once this is merged.

**Review by benPearce1 — APPROVED** [review]
LGTM. Tested locally

**Pull request merged**
